### PR TITLE
Run unit tests on multiple supported iOS versions

### PR DESCRIPTION
**Pull request by evil159**
This PR splits Run Unit tests job into build-unit-tests and run-unit tests. Unit tests are run on iOS 13.7, 14.5, 15.5, and 16.1.

| Before | After |
| ----- | ----- |
|  |  |

## Pull request checklist:
 - [x] Describe the changes in this PR, especially public API changes.
 - [x] Include before/after visuals or gifs if this PR includes visual changes.

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -519,38 +553,61 @@ jobs:
...
-          name: Converting and uploading coverage
-          command: |
-            pip3 install awscli gitpython
-            make update-codecov-with-profdata SCHEME=MapboxMaps BUILD_DIR="$DERIVED_DATA_PATH"
-      - store-logs:
-          artifact_name: MapboxMaps
-          derived_data_path: $DERIVED_DATA_PATH
-      - report-failure:
-          report_failure: << parameters.report_failure >>
-          message: "unit-test-sdk"
```
Comment (OdNairy) [inline .circleci/config.yml:553]: We also have to report failures

File: .circleci/config.yml [hunk 2]
```diff
@@ -519,38 +553,61 @@ jobs:
...
-          path: build/reports
+          name: Run MapboxMaps unit test on << parameters.destination >>  simulator
+          command: |
+            XCTESTRUN_PATH=$(find build -type f -name '*.xctestrun')
+            xcodebuild \
+              test-without-building \
+              -xctestrun "$XCTESTRUN_PATH" \
+              -destination 'platform=iOS Simulator,<< parameters.destination >>' \
+              -enableCodeCoverage YES \
+              -resultBundlePath MapboxMapsTests.xcresult | xcpretty --report junit --output junit.xml
```
Comment (OdNairy) [inline .circleci/config.yml:600]: Can you extract logs to the separate file using `tee` and store them in artifacts

File: .circleci/config.yml [hunk 3]
```diff
@@ -1425,4 +1482,4 @@ commands:
               --url https://api.github.com/repos/mapbox/turf-swift/statuses/<< parameters.turf-revision >> \
               --header 'Accept: application/vnd.github.v3+json' \
               --header "Authorization: token $(mbx-ci github notifier token)" \
-              --data "{\"state\":\"failure\",\"target_url\":\"$CIRCLE_BUILD_URL\",\"description\":\"MapboxMaps SDK integration tests failed\",\"context\":\"Maps SDK Integration\"}"
+              --data "{\"state\":\"failure\",\"target_url\":\"$CIRCLE_BUILD_URL\",\"description\":\"MapboxMaps SDK integration tests failed\",\"context\":\"Maps SDK Integration\"}"
```
Comment (OdNairy) [inline .circleci/config.yml:1485]: Nit: we need a newline at the end of the file

File: .circleci/config.yml [hunk 4]
```diff
@@ -519,38 +553,61 @@ jobs:
...
+            - << parameters.configuration >>-iphonesimulator
+            - "*.xctestrun"
+
+  run-unit-tests:
+    parameters:
+      destination:
+        type: string
+      xcode:
+        type: string
+        default: "13.1.0"
```
Comment (OdNairy) [inline .circleci/config.yml:584]: I think it's better to remove default value as we don't have one in `destination` parameter

File: .circleci/config.yml [hunk 5]
```diff
@@ -519,38 +553,61 @@ jobs:
...
           path: MapboxMapsTests.xcresult.zip
       - store_artifacts:
           path: ~/Library/Logs/DiagnosticReports
-      - locate-derived-data-directory:
-          base_name: $(basename $(pwd))
-      - run:
-          name: Converting and uploading coverage
-          command: |
-            pip3 install awscli gitpython
-            make update-codecov-with-profdata SCHEME=MapboxMaps BUILD_DIR="$DERIVED_DATA_PATH"
```
[the remark this answers is not in the document]
Comment (evil159) [reply .circleci/config.yml:547]: I'm having issues generating code coverage reports with build and run tests steps being separated. I'll address this separately.

**Pull request merged**